storage/imxrt-flash/flashsrv.c

```diff
@@ -12,78 +12,290 @@
+		return -EINVAL;
+
+	return flash_directBytesWrite(&flash_memory->ctx, addr, buff, bufflen);
+}
+
+
+static size_t flashsrv_mfsReadf0(unsigned int addr, void *buff, size_t bufflen)
+{
+	return flashsrv_read(0, addr, buff, bufflen);
+}
+
+
+static size_t flashsrv_mfsReadf1(unsigned int addr, void *buff, size_t bufflen)
+{
+	return flashsrv_read(1, addr, buff, bufflen);
+}
+
+
+static void flashsrv_mfsEraseSectorf0(unsigned int addr)
+{
+	flashsrv_eraseSector(0, addr);
+}
+
+
+static void flashsrv_mfsEraseSectorf1(unsigned int addr)
+{
+	flashsrv_eraseSector(1, addr);
+}
+
+
+static void flashsrv_mfsPartitionErasef0(void)
 {
 	int i;
-	for (i = 0; i < FLASHES_NO; ++i) {
-		if (oid.id == flashsrv_common.flash_memories[i].oid.id)
-			return flashsrv_common.flash_memories + i;
-	}
-	return NULL;
+	uint32_t sectorsNb;
+	flash_memory_t *flash_memory;
+
+	flash_memory = flashsrv_common.flash_memories + 0;
+	sectorsNb = flash_memory->parts[flash_memory->currPart].sz / flash_memory->ctx.properties.sector_size;
+
+	for (i = 0; i < sectorsNb; ++i)
+		flashsrv_eraseSector(0, flash_memory->parts[flash_memory->currPart].offs + flash_memory->ctx.properties.sector_size * i);
 }
 
 
-static void flashsrv_devCtl(msg_t *msg)
+static void flashsrv_mfsPartitionErasef1(void)
+{
+	int i;
+	uint32_t sectorsNb;
+	flash_memory_t *flash_memory;
+
+	flash_memory = flashsrv_common.flash_memories + 1;
+	sectorsNb = flash_memory->parts[flash_memory->currPart].sz / flash_memory->ctx.properties.sector_size;
+
+	for (i = 0; i < sectorsNb; ++i)
+		flashsrv_eraseSector(1, flash_memory->parts[flash_memory->currPart].offs + flash_memory->ctx.properties.sector_size * i);
+}
+
+
+static void flashsrv_devCtl(flash_memory_t *memory, msg_t *msg)
 {
-	flash_memory_t *flash_memory = NULL;
 	flash_i_devctl_t *idevctl = (flash_i_devctl_t *)msg->i.raw;
 	flash_o_devctl_t *odevctl = (flash_o_devctl_t *)msg->o.raw;
 
+	if (memory == NULL) {
+		odevctl->err = -EINVAL;
+		return;
+	}
+
 	switch (idevctl->type) {
 		case flashsrv_devctl_properties:
 			TRACE("imxrt-flashsrv: flashsrv_devctl_properties, id: %d, port: %d.", idevctl->oid.id, idevctl->oid.port);
-			flash_memory = flashsrv_findFlash(idevctl->oid);
-			if (flash_memory == NULL) {
-				odevctl->err = -EINVAL;
-				break;
-			}
+			odevctl->properties.size = memory->ctx.properties.size;
+			odevctl->properties.psize = memory->ctx.properties.page_size;
+			odevctl->properties.ssize = memory->ctx.properties.sector_size;
+			odevctl->properties.offs = 0;
+			odevctl->err = EOK;
+			break;
+
+		case flashsrv_devctl_sync:
+			TRACE("imxrt-flashsrv: flashsrv_devctl_sync, id: %d, port: %d.", idevctl->oid.id, idevctl->oid.port);
+			flash_sync(&memory->ctx);
+			odevctl->err = EOK;
+			break;
+
+		case flashsrv_devctl_eraseSector:
+			TRACE("imxrt-flashsrv: flashsrv_devctl_eraseSector - adrr: %d, id: %d, port: %d.", idevctl->erase.addr, idevctl->oid.id, idevctl->oid.port);
+			flashsrv_eraseSector(memory->fOid.id, idevctl->erase.addr);
```

Comment:
You get flash_memory_t  in flashsrv_devCtl then you get fOid.id from it and then you get flash_memory_t again inside flashsrv_eraseSector. It can be simplified ;)